### PR TITLE
[data] [docs] Datastream docs rename [5/n]

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Part 5 of https://github.com/ray-project/ray/issues/34235

### CODE REVIEW

**Comment by ollie-iterators** [comment]
This is part 5

**Review by matthewdeng — APPROVED** [review]

**Review by scottjlee — APPROVED** [review]

File: doc/source/data/creating-datastreams.rst [hunk 1]
```diff
@@ -801,7 +801,7 @@ Call :func:`~ray.data.read_sql` to read data from a database that provides a
 
         pip install google-cloud-bigquery
 
-    Then, define your connection login and query the dataset.
+    Then, define your connection login and query the datastream.
```
[the remark this answers is not in the document]
Comment (ericl) [reply doc/source/data/creating-datastreams.rst]: Fixed, thanks!

File: doc/source/data/creating-datastreams.rst [hunk 2]
```diff
@@ -504,102 +504,102 @@ futures.
...
 
 .. tabbed:: TensorFlow
 
-    If you already have a TensorFlow dataset available, you can create a Ray Dataset
+    If you already have a TensorFlow dataset available, you can create a Datastream
     using :class:`~ray.data.from_tf`.
 
     .. warning::
         :class:`~ray.data.from_tf` doesn't support parallel reads. You
-        should only use this function with small datasets like MNIST or CIFAR.
```
[the remark this answers is not in the document]
Comment (ericl) [reply doc/source/data/creating-datastreams.rst:551]: I think this should be fixed.

**Review by gjoliver — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Review by scv119 — APPROVED** [review]

**Review by amogkam — APPROVED** [review]

File: doc/source/data/data.rst [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+..
+  https://docs.google.com/drawings/d/16AwJeBNR46_TsrkOmMbGaBK7u-OPsf_V8fHjU-d2PPQ/edit
+
+-------------------------
+Streaming Batch Inference
+-------------------------
+
+Ray Data simplifies general purpose parallel GPU and CPU compute in Ray through its
+powerful :ref:`Datastream <datastream_concept>` primitive. Datastreams enables workloads such as 
+:ref:`GPU batch inference <ref-use-cases-batch-infer>` efficiently on large datasets. Ray Data manages
```
Comment (c21) [inline doc/source/data/data.rst:37]: Should we change `large datasets` to `large input data`?

File: doc/source/data/key-concepts.rst [hunk 4]
```diff
@@ -4,108 +4,103 @@
...
 
-.. image:: images/dataset-shuffle.svg
+.. image:: images/datastream-shuffle.svg
    :align: center
 
 ..
   https://docs.google.com/drawings/d/132jhE3KXZsf29ho1yUdPrCHB9uheHBWHJhDQMXqIVPA/edit
 
-Datasets can shuffle hundreds of terabytes of data. For an in-depth guide on shuffle performance, read :ref:`Performance Tips and Tuning <shuffle_performance_tips>`.
+Datastream can shuffle multi-terabyte datasets, leveraging the Ray object store for disk spilling. For an in-depth guide on shuffle performance, read :ref:`Performance Tips and Tuning <shuffle_performance_tips>`.
```
Comment (c21) [inline doc/source/data/key-concepts.rst:81]: why `datasets` here is better than `data` before?

**Review by c21 — APPROVED** [review]
Still seeing several `Dataset` in https://anyscale-ray--34512.com.readthedocs.build/en/34512/data/examples/nyc_taxi_basic_processing.html and  https://anyscale-ray--34512.com.readthedocs.build/en/34512/data/examples/ocr_example.html .

Are we planning to change them in a followup PR?

File: doc/source/data/data.rst (hunk 3 above)
Comment (ericl) [reply doc/source/data/data.rst:37]: I think this is fine. Lower-case datasets just refers to datasets in general, not the Dataset class.

File: doc/source/data/key-concepts.rst (hunk 4 above)
Comment (ericl) [reply doc/source/data/key-concepts.rst:81]: Lower case 'datasets' is fine to use; this was just an incidental fix to improve the wording here.

**Comment by ericl** [comment]
> Still seeing several Dataset in https://anyscale-ray--34512.com.readthedocs.build/en/34512/data/examples/nyc_taxi_basic_processing.html and https://anyscale-ray--34512.com.readthedocs.build/en/34512/data/examples/ocr_example.html .

Fixed.

**Pull request merged**